src/daria/utils/segmentation.py

```diff
@@ -0,0 +1,229 @@
+    labels = _reset_labels(labels)
+    labels = _fill_holes(labels)
+    labels = _reset_labels(labels)
+    labels = _dilate_by_size(labels, dilation_size, True)
+    labels = _reset_labels(labels)
+    labels = _boundary(labels, boundary_size)
+
+    if verbosity:
+        plt.figure("Final result after clean up")
+        plt.imshow(labels)
+        plt.show()
+
+    return labels
+
+
+# ! ---- Auxiliary functions for segment
+
+
+def _reset_labels(labels: np.ndarray) -> np.ndarray:
+    """
+    Rename labels, such that these are consecutive with step size 1,
+    starting from 0.
+
+    Args:
+        labels (np.ndarray): labeled image
+
+    Returns:
+        np.ndarray: new labeled regions
+    """
+    pre_labels = np.unique(labels)
+    for i, label in enumerate(pre_labels):
+        mask = labels == label
+        labels[mask] = i
+    return labels
+
+
+def _fill_holes(labels: np.ndarray) -> np.ndarray:
+    """
+    Routine for filling holes in all labeled regions.
+
+    Args:
+        labels (np.ndarray): labeled image
+
+    Returns:
+        np.ndarray: labels without holes.
+    """
+    pre_labels = np.unique(labels)
+    for label in pre_labels:
+        mask = labels == label
+        mask = ndi.binary_fill_holes(mask).astype(bool)
+        labels[mask] = label
+    return labels
+
+
+def _dilate_by_size(
+    labels: np.ndarray, footprint: np.ndarray, decreasing_order: bool
+) -> np.ndarray:
+    """
+    Dilate objects by prescribed size.
+
+    Args:
+        labels (np.ndarray): labeled image
+        footprint (np.ndarray): foot print for dilation
+        descreasing_order (bool): flag controlling whether dilation
+            should be performed on objects with decreasing order
+            or not (increasing order then).
+
+    Returns:
+        np.ndarray: labels after dilation
+    """
+
+    # Determine sizes of all marked areas
+    pre_labels = np.unique(labels)
+    sizes = [np.count_nonzero(labels == label) for label in pre_labels]
+    # Sort from small to large
+    labels_sorted_sizes = np.argsort(sizes)
+    if decreasing_order:
+        labels_sorted_sizes = np.flip(labels_sorted_sizes)
+    # Erode for each label if still existent
+    for label in labels_sorted_sizes:
+        mask = labels == label
+        mask = skimage.morphology.binary_dilation(
+            mask, skimage.morphology.disk(footprint)
+        )
+        labels[mask] = label
+    return labels
+
+
+def _boundary(labels: np.ndarray, thickness: int) -> np.ndarray:
+    """
+    Constant extenion in normal direction at the boundary of labeled image.
+
+    Args:
+        labels (np.ndarray): labeled image
+        thickness (int): thickness of boundary which should be overwritten
+
+    Returns:
+        np.ndarray: updated labeled image
+    """
+    # Top
+    labels[:thickness, :] = labels[thickness : thickness + 1, :]
+    # Bottom
+    labels[-thickness - 1 : -1, :] = labels[-thickness - 1 : -thickness, :]
```

Comment:
```suggestion
    labels[-thickness :, :] = labels[-thickness - 1 : -thickness, :]
```
Include the last row.